solr/core/src/java/org/apache/solr/util/circuitbreaker/CircuitBreaker.java

```diff
@@ -62,6 +64,16 @@ public CircuitBreaker() {}
   /** Get error message when the circuit breaker triggers */
   public abstract String getErrorMessage();
 
+  /**
+   * Get http error code, defaults to {@link SolrException.ErrorCode#TOO_MANY_REQUESTS} but can be
+   * overridden with system property {@link #SYSPROP_SOLR_CIRCUITBREAKER_ERRORCODE}
+   */
+  public static SolrException.ErrorCode getErrorCode() {
+    return SolrException.ErrorCode.getErrorCode(
+        Integer.getInteger(
+            SYSPROP_SOLR_CIRCUITBREAKER_ERRORCODE, SolrException.ErrorCode.TOO_MANY_REQUESTS.code));
```

Comment:
edge case: wondering about noisily failing if the property is specified but invalid e.g. `-Dsolr.circuitbreaker.errorcode=SERVICE_UNAVAILABLE` or some such misunderstanding.

Reply:
You'd get an exception from either the `getInteger` call or the `getErrorCode` call. But only during runtime. We could of course have a static context initialization, but I put init here mainly to be able to write a test...

Reply:
So in this case the end user / client would receive HTTP 500 error messages from the exception bubbling up I suppose. I could add a `getErrorCode()` call to the constructor to fail during initialization too?

Reply:
Sorry, I missed this comment. The exception bubbling up could be confusing for clients and operationally an exception or at least logging on startup could be helpful?

Reply:
So I learnt something: `Integer.getInteger()` will return the default-value also if the system property is set but is not an integer. And `getErrorCode` will return `UNKNOWN` (0) for unsupported codes such as `999`. To fail loudly we must catch both, and I'll push a solution.